### PR TITLE
Add missing defaultWeekSettings property type to Settings class

### LINKED ISSUE #68080: [@types/luxon] new `defaultWeekSettings` type missing for `Settings.js`
### Discussed in https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/68079

<div type='discussions-op-text'>

<sup>Originally posted by **hokwanhung** January  3, 2024</sup>
As asked in moment/luxon#1561, there seems to be a missing `type` for the newly added property `defaultWeekSettings`.

**To Reproduce**
[Edit luxon-defaultWeekSettings-error-with-TypeScript](https://codesandbox.io/p/devbox/luxon-defaultweeksettings-error-with-typescript-fsy4gm?embed=1&file=%2Fyarn.lock)</div>

### PR DESCRIPTION
**Pull request by timagixe**
Fixes https://github.com/DefinitelyTyped/DefinitelyTyped/issues/68080

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [source code](https://github.com/moment/luxon/blob/825b77a6cc16a75d6e8609c484a61c2ffe66b845/src/settings.js#L113-L129), [WeekSettings in documentation](https://moment.github.io/luxon/api-docs/index.html#weeksettings), [defaultWeekSettings property in documentation](https://moment.github.io/luxon/api-docs/index.html#settingsdefaultweeksettings)

### CODE REVIEW

File: types/luxon/src/weekSettings.d.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { WeekdayNumbers } from "./datetime";
+
+export interface WeekSettings {
```
Comment (CarsonF) [inline types/luxon/src/weekSettings.d.ts:3]: I would just keep this in the settings file. It's not really applicable anywhere else currently and the js src doesn't have this file.

File: types/luxon/src/settings.d.ts [hunk 2]
```diff
@@ -65,6 +66,15 @@ export class Settings {
...
     static resetCaches(): void;
+
+    static get defaultWeekSettings(): WeekSettings | null;
+
+    /**
+     * Allows overriding the default locale week settings, i.e. the start of the week, the weekend and
+     * how many days are required in the first week of a year.
+     * Does not affect existing instances.
+     */
+    static set defaultWeekSettings(weekSettings: WeekSettings | null);
```
Comment (CarsonF) [inline types/luxon/src/settings.d.ts:77]: These could just be the property
```suggestion
    /**
     * Allows overriding the default locale week settings, i.e. the start of the week, the weekend and
     * how many days are required in the first week of a year.
     * Does not affect existing instances.
     */
    static defaultWeekSettings: WeekSettings | null;
```

**Review by CarsonF — CHANGES_REQUESTED** [review]
Thanks @timagixe!

File: types/luxon/src/weekSettings.d.ts (hunk 1 above)
Comment (timagixe) [reply types/luxon/src/weekSettings.d.ts]: Landed in [7828fd4](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/68088/commits/7828fd4198e62d415792b7abdc7218154e5074fd)

**Review by CarsonF — APPROVED** [review]

File: types/luxon/src/settings.d.ts (hunk 2 above)
Comment (timagixe) [reply types/luxon/src/settings.d.ts]: Landed in [ee29728](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/68088/commits/ee297285c6d37b3165c13ccaf37f8c9c37b9cc26)

Thank you for the review! 🙌

**Review by peterblazejewicz — APPROVED** [review]
@timagixe thanks!

**Comment by timagixe** [comment]
Ready to merge

**Pull request merged**